### DISCUSSION TITLE
How do you cancel a delayed job?

### DISCUSSION THREAD

**Discussion by foremtehan** [laravel/framework#51972] [Ideas]
I know we can find the details in the payload, but this approach is messy. Is there a better way? For example, if you want to delete a job that sends an email to a user with an ID of 10 in one hour but you want to cancel this operation

**Comment by ghost** [comment]
Hi. Good question. First thought that comes to mind is a new table in db that stores the deleted job identifier. Then before executing each job, look there to see if the job is still valid.

But this is not the way it should be...

You should read about you queue driver (sqs mysql etc) and see if it supports this feature. If yes, then you should implement it.
